crates/sui/Cargo.toml

```diff
@@ -119,3 +130,6 @@ gas-profiler = [
     "sui-types/gas-profiler",
     "sui-execution/gas-profiler",
 ]
+default = ["postgres-feature"]
+postgres-feature = ["diesel/postgres", "diesel/postgres_backend"]
+pg_integration = []
```

Comment:
The existence of both these features (`postgres-feature` and `pg_integration`) is interesting (the latter seems to not be hooked up at all at the moment, at least in the toml file) -- it seems like there should be just one for `sui` that controls everything.

Also, is it an issue that we now expect postgres to be available by default? I.e. will this bring the issues that people were facing while trying to build `sui-test-validator` over to `sui`?

Reply:
Yes, in this PR I have not focused on how to deal with the PG stuff. Also, those postgres-feature and pg_integration fields might be useless, I will revise those! 

In a nutshell, we have two options: 
1) put the `with-indexer` and `with-graphql` under a cfg option. The problem here is that users will now have to build from source rather than using the binary, unless we decide to build the `sui` binary will full features and release it.
2) figure out how to bundle the pg library with the binary, statically. We tried hard and we did not manage, but probably this is still the better way such that users do not need to build from scratch. 

It's just a tradeoff.

Reply:
I like the sound of option (1) (the second variant) -- put indexer and GraphQL support under a feature flag (the same one), but enable them in the downloadable build we offer (and from homebrew and choco).

We can even make it so the feature is enabled by default, but by having them under a feature flag, we make it so that if someone wants to build the `sui` CLI from scratch and they either don't want to deal with PG or they are having some trouble with it, then they can build with that flag disabled to get unblocked. The thing I am otherwise nervous about is that we expose people who are used to using the `sui` CLI to this additional dependency/linking complexity and they don't have a way out even if they don't care to use the indexer/GraphQL.

Long-term, the move here is to not require PG support to use indexer/GraphQL with localnet (e.g. sqlite support or similar).

Reply:
I put the indexer and graphql behind a cfg option. 

Unfortunately, I don't think releasing this binary with full features is possible as it will break any installation that does not have libpq. Particularly, if anyone just did `brew install sui`, then they will not have had access to `sui-test-validator` and thus if this `postgres` feature is enabled by default, the binary will error.

Reply:
The alternative is to have two binaries built, but that's also kinda stupid.
- sui
- sui-pg - which needs libpq and postgres to work with `--with-indexer` and `--with-graphql`.